<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="gglogo">
<title>Considerations for re-designing the traditional logo sequence plot • gglogo</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Considerations for re-designing the traditional logo sequence plot">
<meta property="og:description" content="gglogo">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gglogo</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.5</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Design_considerations.html">Considerations for re-designing the traditional logo sequence plot</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/heike/gglogo/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Considerations for re-designing the traditional logo sequence plot</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/heike/gglogo/blob/HEAD/vignettes/Design_considerations.Rmd" class="external-link"><code>vignettes/Design_considerations.Rmd</code></a></small>
      <div class="d-none name"><code>Design_considerations.Rmd</code></div>
    </div>

    
    
<p>In a traditional logo sequence plot <span class="citation">[@logo]</span>, sequences of nucleotides or amino acids
are summarized by creating a frequency break-down of letters used in
each position across the sequences (using the assumption that the
sequences are aligned, and have all the same length <span class="math inline">\(L\)</span>).</p>
<p>The Shannon information <span class="math inline">\(I\)</span> <span class="citation">[@shannon]</span> is used as a measure of entropy (in
bits) to describe the amount of mixture in each position.</p>
<p>Let all sequences have length <span class="math inline">\(L\)</span>,
with an alphabet <span class="math inline">\(A\)</span> of size <span class="math inline">\(K\)</span>. The alphabet is the set of all
symbols/characters in a sequence. For nucleotides, the alphabet <span class="math inline">\(A\)</span> has <span class="math inline">\(K=4\)</span> elements, and generally consists of
the set of letters {A, C, G, T} (standing for <strong>A</strong>denine,
<strong>C</strong>ystosine, <strong>T</strong>hymine) for DNA or the
letters {A, C, G, U}, for RNA (<strong>U</strong>racil replaces
thymine).</p>
<p>In the case of peptide sequences, each letter represents one of 21
amino acids (see e.g. <a href="https://en.wikipedia.org/wiki/DNA_and_RNA_codon_tables#/media/File:Aminoacids_table.svg" class="external-link">Wikipedia’s
codon chart</a>).</p>
<p>Let <span class="math inline">\(f_a(p)\)</span> be the (relative)
frequency, with which letter <span class="math inline">\(a \in
A\)</span> is observed in position <span class="math inline">\(p\)</span>, <span class="math inline">\(1 \le p
\le L\)</span> of a set of sequences of length <span class="math inline">\(L\)</span>.</p>
<p>The <strong>Shannon conservation index</strong> <span class="math inline">\(I\)</span> in position <span class="math inline">\(p\)</span> is defined as <span class="math display">\[
I(p) := \log_2 (K) - \sum_{a \ \in \ A} f_{a}(p)
\log_2\left(f_{a}(p)\right).
\]</span> By defining the expression <span class="math inline">\(f
\log_2 (f) := 0\)</span> for <span class="math inline">\(f = 0\)</span>,
<span class="math inline">\(I(p)\)</span> is well defined for all
frequencies <span class="math inline">\(f \in [0,1]\)</span>. The
measures reaches a minimum of 0 when <span class="math inline">\(f_a =
1/K\)</span> for all <span class="math inline">\(a \in A\)</span>, while
a maximum of <span class="math inline">\(\log_2 (K)\)</span> is reached
when all frequencies <span class="math inline">\(f_a\)</span> are 0
except for one <span class="math inline">\(a_0\)</span> with <span class="math inline">\(f_{a_0}=1\)</span>. (Note that the second term is
a measure of information/entropy - depending on the choice of the base
of the logarithm result in differently named entropy measures: base
<span class="math inline">\(e\)</span> results in the natural entropy
measured in ‘nats’, while base 10 is measured in ‘dits’).</p>
<p>For 21 amino acids, the maximal conservation is <span class="math inline">\(-\log_2 (1/21) = 4.39\)</span> bits, which is
reached, if only a single amino acid is observed in the position, while
perfect diversity/minimal conservation of 0 bits is reached, when all 21
amino acids are observed with the same frequency.</p>
<p>In the traditional logo sequence plot, a set of sequences is
summarised, by scaling the heights of the letters corresponding to each
amino acids (letter in the alphabet) by their contribution to a
position’s total conservation. The letters are then stacked by size,
with the amino acid of the largest contribution on top.</p>
<div class="section level2">
<h2 id="shortcomings-of-the-traditional-logo-plot">Shortcomings of the traditional logo plot<a class="anchor" aria-label="anchor" href="#shortcomings-of-the-traditional-logo-plot"></a>
</h2>
<p>The <strong>color choice</strong> for representing the letters of
amino acids is related to water solubility and polarity
(e.g. hydrophobic, non-polar amino acids are shown in red) but this is
not explicitly stated in a legend. Further, the use of letters to
represent amino acids results in shapes of different visual dominance;
the letter <code>I</code> is much less visually pronounced than for
example <code>W</code>.</p>
<p><strong>Non identified amino acids</strong> are being ignored in the
original plot – it is of importance to keep track of at least the
position and the frequency of these occurrences, as it might indicate a
problem with the sequencing.</p>
<p>The plotting of <strong>sequences of subfamilies</strong> in separate
logo plots does not facilitate a comparison of them. Researchers are in
particular interested in differences in the conservation of amino acids
between subfamilies.</p>
<p>The <strong>number of sequences</strong> in each of the subfamilies
is not shown directly. This influences the inherent variability. It is
therefore important to keep track of these numbers to be able to assess
how and whether the size of each subfamily affects conservation.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/heike/gglogo" class="external-link">gglogo</a></span><span class="op">)</span></span></code></pre></div>
</div>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Heike Hofmann, Eric Hare, GGobi Foundation.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
